pydoctor/test/test_astbuilder.py

```diff
@@ -71,17 +71,37 @@ def unwrap(parsed_docstring: ParsedEpytextDocstring) -> str:
     assert isinstance(value, str)
     return value
 
+class NotFoundLinker(DocstringLinker):
+    """A DocstringLinker implementation that cannot find any links."""
+
+    def resolve_identifier(self, identifier: str) -> Optional[str]:
+        return None
+
+    def resolve_identifier_xref(self, identifier: str, lineno: int) -> str:
+        raise LookupError(identifier)
+
 def to_html(parsed_docstring: ParsedDocstring) -> str:
-    return flatten(parsed_docstring.to_stan(None))
+    return flatten(parsed_docstring.to_stan(NotFoundLinker()))
 
-def type2str(type_expr: object) -> Optional[str]:
+@overload
+def type2str(type_expr: None) -> None: ...
+
+@overload
+def type2str(type_expr: ast.expr) -> str: ...
+
```

Comment:
I am not sure why you don't write the implementation when you declare the overloading declaration and remove the definition without the overload decorator

```suggestion
@overload
def type2str(type_expr: None) -> None:
    return None

@overload
def type2str(type_expr: ast.expr) -> str:
    src = astor.to_source(type_expr)
    assert isinstance(src, str)
    return src.strip()
```

Reply:
[`overload`](https://docs.python.org/3/library/typing.html#typing.overload) doesn't work like that: the overloaded variants cannot be called, they're only for type checking. What you're thinking of is [`singledispatch`](https://docs.python.org/3/library/functools.html#functools.singledispatch), but I don't think the extra complexity of that is worth it when we only have 2 variants.

Reply:
my bad... this is `typing.overload` and not https://www.python.org/dev/peps/pep-3124/ .. .and I see that PEP-3124 was not implemented.

thanks for the info.